__tests__/__unit__/shared/actions.unit.test.ts

```diff
@@ -35,41 +39,43 @@ async function createGlobalMocks() {
     Object.defineProperty(globals, "LOG", { value: jest.fn(), configurable: true });
     Object.defineProperty(globals.LOG, "debug", { value: jest.fn(), configurable: true });
     Object.defineProperty(globals.LOG, "error", { value: jest.fn(), configurable: true });
+
+    return globalMocks;
 }
 
 // Idea is borrowed from: https://github.com/kulshekhar/ts-jest/blob/master/src/util/testing.ts
 const mocked = <T extends (...args: any[]) => any>(fn: T): jest.Mock<ReturnType<T>> => fn as any;
 
 describe("Shared Actions Unit Tests - Function searchForLoadedItems", () => {
-    let blockMocks;
-
     function createBlockMocks() {
-        const session = createISessionWithoutCredentials();
-        const treeView = createTreeView();
-        const imperativeProfile = createIProfile();
-        const profileInstance = createInstanceOfProfile(imperativeProfile);
-        const datasetSessionNode = createDatasetSessionNode(session, imperativeProfile);
+        const newMocks = {
+            session: createISessionWithoutCredentials(),
+            treeView: createTreeView(),
+            imperativeProfile: createIProfile(),
+            profileInstance: null,
+            datasetSessionNode: null,
+            ussSessionNode: null,
+            testDatasetTree: null,
+            testUssTree: null
+        };
 
+        newMocks.profileInstance = createInstanceOfProfile(newMocks.imperativeProfile);
         // It's required to have proper mock of profile for USS Node generation
-        mocked(Profiles.getInstance).mockReturnValue(profileInstance);
-        const ussSessionNode = createUSSSessionNode(session, imperativeProfile);
-
-        return {
-            session,
-            imperativeProfile,
-            profileInstance,
-            datasetSessionNode,
-            ussSessionNode,
-            testDatasetTree: createDatasetTree(datasetSessionNode, treeView),
-            testUssTree: createUSSTree([], [ussSessionNode], treeView)
-        };
+        mocked(Profiles.getInstance).mockReturnValue(newMocks.profileInstance);
+        newMocks.datasetSessionNode = createDatasetSessionNode(newMocks.session, newMocks.imperativeProfile);
+        newMocks.ussSessionNode = createUSSSessionNode(newMocks.session, newMocks.imperativeProfile);
+        newMocks.testUssTree = createUSSTree([], [newMocks.ussSessionNode], newMocks.treeView);
+        newMocks.testDatasetTree = createDatasetTree(newMocks.datasetSessionNode, newMocks.treeView);
+
+        return newMocks;
     }
```

Comment:
I do not understand this change. Before we had constants one building on top of another and it was easy to follow. Now we created an object and we are mutating it.

Reply:
The rest of the PR looks good, but could we use the previous style in this function?

Reply:
I have been creating and mutating an object containing the mock variables in all of my testing PRs. I don't really see why this is problem. Inside the it() block the blockMocks variable is sometimes mutated as well.

Reply:
Code that is mutating stuff is harder to read for people who did not see the code before because they have to go back and forth to what has been mutated when and in what order. Building up constants from constants on the other hand is straightforward to follow.

I must have overlooked the mutations before.

Reply:
For the reasoning behind this I think an old [Dijkstra quote](https://en.wikiquote.org/wiki/Edsger_W._Dijkstra)  explains this well

> Our intellectual powers are rather geared to master static relations and ... our powers to visualize processes evolving in time are relatively poorly developed. For that reason we should do (as wise programmers aware of our limitations) our utmost to shorten the conceptual gap between the static program and the dynamic process, to make the correspondence between the program (spread out in text space) and the process (spread out in time) as trivial as possible.

Reply:
This code is small, so it is not a bid deal. But I would like us to think how to make the code simpler and easier to follow. Avoiding mutations when possible is one way to do that.